### PR TITLE
docs: misc docs and tests (MINOR)

### PR DESCRIPTION
**Pull request by agavra**
### Description 

- add documentation around string constants
- add some test coverage to PARTITION BY with quoted identifiers

### Testing done 

- ran the QTT test

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: docs-md/developer-guide/syntax-reference.md [hunk 1]
```diff
@@ -312,6 +312,20 @@ Mathematical operations between `DOUBLE` and `DECIMAL` cause the decimal to be
 converted to a double value automatically. Converting from the decimal data type
 to any floating point type (`DOUBLE`) may cause loss of precision.
 
+### Constants
+
+- **String constants** are enclosed in single quotation marks and may include any unicode
+character (e.g. `'hello'`, `'1.2'`)
```
Comment (JimGalasyn) [inline docs-md/developer-guide/syntax-reference.md]:
```suggestion
character (e.g. `'hello'`, `'1.2'`).
```

File: docs-md/developer-guide/syntax-reference.md [hunk 2]
```diff
@@ -312,6 +312,20 @@ Mathematical operations between `DOUBLE` and `DECIMAL` cause the decimal to be
...
+- **Integer constants** are represented by numbers that are not enclosed in quotation marks
+and do not contain decimal points (e.g. `1`, `2`)
```
Comment (JimGalasyn) [inline docs-md/developer-guide/syntax-reference.md]:
```suggestion
and do not contain decimal points (e.g. `1`, `2`).
```

File: docs-md/developer-guide/syntax-reference.md [hunk 3]
```diff
@@ -312,6 +312,20 @@ Mathematical operations between `DOUBLE` and `DECIMAL` cause the decimal to be
...
+- **Decimal constants** are represented by a string of numbers that are no enclosed in quotation
+marks and contain a decimal point (e.g. `1.2`, `87.`, `.94`). The type of the decimal constant
+will be `DECIMAL(p, s)` where `p` is the total number of numeric characters in the string and
+`s` is the total number of numeric characters that appear to the right of the decimal point.
+- **Double constants** are numeric strings represented in scientific notation (e.g. `1E0`, `.42E-3`)
```
Comment (JimGalasyn) [inline docs-md/developer-guide/syntax-reference.md]:
```suggestion
- **Double constants** are numeric strings represented in scientific notation (e.g. `1E0`, `.42E-3`).
```

File: docs-md/developer-guide/syntax-reference.md [hunk 4]
```diff
@@ -312,6 +312,20 @@ Mathematical operations between `DOUBLE` and `DECIMAL` cause the decimal to be
...
+- **Boolean constants** are the unquoted strings that are exactly (case-insensitive) `TRUE` 
+or `FALSE`
```
Comment (JimGalasyn) [inline docs-md/developer-guide/syntax-reference.md]:
```suggestion
or `FALSE`.
```

**Comment by agavra** [comment]
Thanks for the review @JimGalasyn  - my docs aren't accurate to 5.4, I wrote them specifically to reflect #4335

File: docs-md/developer-guide/syntax-reference.md [hunk 5]
```diff
@@ -312,6 +312,20 @@ Mathematical operations between `DOUBLE` and `DECIMAL` cause the decimal to be
...
+- **Integer constants** are represented by numbers that are not enclosed in quotation marks
+and do not contain decimal points (e.g. `1`, `2`).
+- **Decimal constants** are represented by a string of numbers that are no enclosed in quotation
+marks and contain a decimal point (e.g. `1.2`, `87.`, `.94`). The type of the decimal constant
+will be `DECIMAL(p, s)` where `p` is the total number of numeric characters in the string and
+`s` is the total number of numeric characters that appear to the right of the decimal point.
+- **Double constants** are numeric strings represented in scientific notation (e.g. `1E0`, `.42E-3`).
+- **Boolean constants** are the unquoted strings that are exactly (case-insensitive) `TRUE`
+or `FALSE`.
+
```
Comment (big-andy-coates) [inline docs-md/developer-guide/syntax-reference.md]: Might it be worth adding something about the supported implicit conversions of numeric types, e.g. decimal implicitly converts to double etc.

File: docs-md/developer-guide/syntax-reference.md [hunk 6]
```diff
@@ -312,6 +312,20 @@ Mathematical operations between `DOUBLE` and `DECIMAL` cause the decimal to be
 converted to a double value automatically. Converting from the decimal data type
 to any floating point type (`DOUBLE`) may cause loss of precision.
 
+### Constants
+
+- **String constants** are enclosed in single quotation marks and may include any unicode
+character (e.g. `'hello'`, `'1.2'`).
+- **Integer constants** are represented by numbers that are not enclosed in quotation marks
+and do not contain decimal points (e.g. `1`, `2`).
```
Comment (big-andy-coates) [inline docs-md/developer-guide/syntax-reference.md]: Might be worth stating such constants can be of type `INT` or `BIGINT`.

**Pull request merged**